how-to/workspace-platform-starter/client/src/framework/shapes/conditions-shapes.ts

```diff
@@ -18,9 +18,20 @@ export interface Conditions<O = unknown, H = ModuleHelpers> extends ModuleImplem
  */
 export type ConditionsProviderOptions = ModuleList;
 
+/**
+ * Callback that a condition must implement.
+ */
+export type ConditionCallback = (
+	platform: WorkspacePlatformModule,
+	context?: {
+		callerType?: string;
```

Comment:
Should this be an exported type e.g. conditionCallerType = "dock"|"browser"|"menu"|"init-options" etc ?

Reply:
Yes we could limit it, I trying to think if there was a reason we might need it to be freeform string

Reply:
Updated to more strongly type the context objects, both the callerType and the customData